app/screens/home/recent_mentions/index.tsx

```diff
@@ -29,20 +95,106 @@ const RecentMentionsScreen = () => {
             opacity: withTiming(0, {duration: 150}),
             transform: [{translateX: withTiming(toLeft ? -25 : 25, {duration: 150})}],
         };
-    }, [start]);
+    }, [isFocused]);
+
+    const renderItem = useCallback(({item}) => {
+        if (typeof item === 'string') {
+            if (isDateLine(item)) {
+                return (
+                    <DateSeparator
+                        date={getDateForDateLine(item)}
+                        theme={theme}
+                        timezone={isTimezoneEnabled ? currentTimezone : null}
+                    />
+                );
+            }
+            return null;
+        }
+
+        return (
+            <Mention
+                currentUser={currentUser}
+                post={item}
+            />
+        );
+    }, [currentUser]);
 
     return (
-        <SafeAreaView
-            style={{flex: 1, backgroundColor: 'brown'}}
-        >
-            <Animated.View
-                onLayout={onLayout}
-                style={[{flex: 1, justifyContent: 'center', alignItems: 'center'}, animated]}
+        <>
+            <NavigationHeader
+                isLargeTitle={isLargeTitle}
+                showBackButton={false}
+                subtitle={subtitle}
+                title={title}
+                hasSearch={false}
+                scrollValue={scrollValue}
+                forwardedRef={scrollRef}
+            />
+            <SafeAreaView
+                style={{flex: 1}}
```

Comment:
Probably over optimization, but since this (and next) is an object (or list) and changes in every render, should we make them as constants? (outside the render function)

Reply:
oh, was meaning to do that..!

Reply:
Does it make sense to do something similar for the next line (edges attribute)?

Reply:
done